pcsx2/USB/usb-pad/usb-flightstick.cpp

```diff
@@ -0,0 +1,691 @@
+		data.rudder_right = 0;
+		data.throttle_up = 0;
+		data.throttle_down = 0;
+		data.stick_hat_left = 0;
+		data.stick_hat_right = 0;
+		data.stick_hat_up = 0;
+		data.stick_hat_down = 0;
+
+		data.stick_x = analog_center;
+		data.stick_y = analog_center;
+		data.rudder = analog_center;
+		data.throttle = analog_center;
+		data.hatstick_x = analog_center;
+		data.hatstick_y = analog_center;
+		data.button_a = 0x00;
+		data.button_b = 0x00;
+		data.buttons = 0;
+	}
+
+	void FlightStickDeviceState::UpdateStick() noexcept
+	{
+		if (data.stick_left > 0)
+			data.stick_x = static_cast<u8>(std::max<int>(analog_range - data.stick_left, 0));
+		else if (data.stick_right > 0)
+			data.stick_x = static_cast<u8>(std::min<int>(analog_range + data.stick_right, analog_range * 2));
+		else
+			data.stick_x = 0x80;
+
+		if (data.stick_up > 0)
+			data.stick_y = static_cast<u8>(std::max<int>(analog_range - data.stick_up, 0));
+		else if (data.stick_down > 0)
+			data.stick_y = static_cast<u8>(std::min<int>(analog_range + data.stick_down, analog_range * 2));
+		else
+			data.stick_y = 0x80;
+	}
+	void FlightStickDeviceState::UpdateRudder() noexcept
+	{
+		if (data.rudder_left > 0)
+			data.rudder = static_cast<u8>(std::max<int>(analog_range - data.rudder_left, 0));
+		else if (data.rudder_right > 0)
+			data.rudder = static_cast<u8>(std::min<int>(analog_range + data.rudder_right, analog_range * 2));
+		else
+			data.rudder = 0x80;
+	}
+	void FlightStickDeviceState::UpdateThrottle() noexcept
+	{
+		if (data.throttle_up > 0)
+			data.throttle = static_cast<u8>(std::min<int>(analog_range + data.throttle_up, analog_range * 2));
+		else if (data.throttle_down > 0)
+			data.throttle = static_cast<u8>(std::max<int>(analog_range - data.throttle_down, 0));
+		else
+			data.throttle = 0x80;
+	}
+
+	void FlightStickDeviceState::UpdateStickHat() noexcept
+	{
+		if (data.stick_hat_left > 0)
+			data.hatstick_x = static_cast<u8>(std::max<int>(analog_range - data.stick_hat_left, 0));
+		else if (data.stick_hat_right > 0)
+			data.hatstick_x = static_cast<u8>(std::min<int>(analog_range + data.stick_hat_right, analog_range * 2));
+		else
+			data.hatstick_x = 0x80;
+
+		if (data.stick_hat_up > 0)
+			data.hatstick_y = static_cast<u8>(std::max<int>(analog_range - data.stick_hat_up, 0));
+		else if (data.stick_hat_down > 0)
+			data.hatstick_y = static_cast<u8>(std::min<int>(analog_range + data.stick_hat_down, analog_range * 2));
+		else
+			data.hatstick_y = 0x80;
+	}
+
+	static void flightstick_handle_data(USBDevice* dev, USBPacket* p)
+	{
+		FlightStickDeviceState* s = USB_CONTAINER_OF(dev, FlightStickDeviceState, dev);
+
+		if (p->pid != USB_TOKEN_IN || p->ep->nr != 1)
+		{
+			Console.Error("Unhandled FlightStickController request pid=%d ep=%u", p->pid, p->ep->nr);
+			p->status = USB_RET_STALL;
+			return;
+		}
+
+		switch (s->type)
+		{
+			case FLIGHTSTICK_FS1:
+			case FLIGHTSTICK_FS2:
+			{
+				//interrupt input data
+				FlightStickConData out = {};
```

Comment:
You can store `FlightStickConData_VR00 vendordata_00{};`, `FlightStickConData_VR01 vendordata_01{};` and `FlightStickConData out{};` in `FlightStickDeviceState` and then update these structures directly from `SetBindingValue` to avoid making the forth copy of the same data.